src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/AddressDtoValidator.java

```diff
@@ -93,7 +95,11 @@ private void validateCountry(String parentAddressField, String country, List<Str
         if (countryNotBlank) {
             boolean isOnList = allowedCountries.contains(country);
             if (!isOnList) {
-                var validationMessage = String.format(ValidationMessages.COUNTRY_NOT_ON_LIST_ERROR_MESSAGE, country);
+                String sanitisedCountryName = Encode.forJava(country);
```

Comment:
the sanitise and length truncation are done for a couple of fields so rather than duplicate the code to do that, I would make a new Sanitiser class under utils, have a method 'makeSafeForLogging'? that takes a string, Encodes it and truncates it if > 50 and returns the 'safe' string. That way we can keep the 'make safe' logic in one place and can unit test it. Have it as a @Component that can be injected where it is needed.

Reply:
Refactored and tested this.